presto-base-jdbc/src/main/java/io/prestosql/plugin/jdbc/BaseJdbcClient.java

```diff
@@ -795,32 +807,20 @@ protected String quoted(String catalog, String schema, String table)
 
     protected static Optional<String> escapeNamePattern(Optional<String> name, Optional<String> escape)
     {
-        if (!name.isPresent() || !escape.isPresent()) {
-            return name;
-        }
-        return Optional.of(escapeNamePattern(name.get(), escape.get()));
+        checkArgument(escape.isPresent(), "Escape string is missing");
```

Comment:
We shouldn't pass an `Optional` (which implies it can be missing) and then unconditionally fail when it's not present. Have the caller fail instead (where they potentially have more information).

Reply:
Done